test/acceptance/testrunners/filepermissiontestrunner.go

```diff
@@ -0,0 +1,114 @@
+package testrunners
+
+import (
+	"github.com/aws/amazon-cloudwatch-agent-test/filesystem"
+	"github.com/aws/amazon-cloudwatch-agent-test/internal/rule"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/status"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/test_runner"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"log"
+	"time"
+)
+
+type FilePermissionTestRunner struct {
+	test_runner.BaseTestRunner
+}
+
+var _ test_runner.ITestRunner = (*FilePermissionTestRunner)(nil)
+
+const agentConfigPath = "/opt/aws/amazon-cloudwatch-agent/bin/config.json"
+const agentConfigCopiedDir = "/opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.d"
+const translatedTomlPath = "/opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.toml"
+
+var (
+	onlyRootCanWriteRule = rule.Rule[string]{
+		Conditions: []rule.ICondition[string]{
+			&rule.PermittedEntityMatch{ExpectedOwner: aws.String("root"), ExpectedGroup: aws.String("root")},
```

Comment:
Why are we using `aws.String` here? Do we need to use pointer references to a static string "root"? Even if we do, do we need to use this function call instead of just referencing with `&`?

Reply:
No, when I made my own string pointer I was told to use that in some other PRs so just kept doing it. didn't really care except personally I really don't like leaking things like this into every file

Reply:
I think I was the one that said that, but I meant to use it for constructing the AWS SDK request structs, not everywhere 😄